Add iOS card style guidance (#525)
nhsuk/nhsapp-frontend: docs/ios/card-style.md

---
layout: layouts/ios.njk
title: Card style
tags:
  - iosStyles
---

The `nhsCardStyle()` modifier applies NHS card styling to any view: a padded, rounded background with an optional border.

> Components like the [banner](/ios/banner) and [campaign card](/ios/campaign-card) use it.

Author: No, this URL will point to the iOS campaign card guidance when published.

{% endcall %}

```swift { .nhsuk-code--button }
VStack(alignment: .leading, spacing: 8) {
    Text("Card title")
        .font(.nhsHeadline)
    Text("Some supporting body text inside a card.")
        .font(.nhsBody)
}
.foregroundStyle(.nhsText)
.nhsCardStyle()
```

For an outlined card, use a clear background with a border:

<img src="/assets/images/ios/card-style-outline.png" width="375">

```swift { .nhsuk-code--button }
Text("Card with a border")
    .foregroundStyle(.nhsText)
    .nhsCardStyle(
        backgroundColor: .clear,
        borderWidth: 2
    )
```

To make the whole card tappable, apply the modifier to a `Button`:

```swift { .nhsuk-code--button }
Button {
    // open the details screen
} label: {
    Text("Entire card is tappable")
}
.nhsCardStyle()
```

### Using a card in a list view

To place a card inside a `List`, apply `nhsCardRowStyle()` after `nhsCardStyle()`. This removes the default row insets, background and separator so the card renders edge to edge rather than as a standard inset row:

```swift { .nhsuk-code--button }
List {
    Text("Card in a list")
        .nhsCardStyle()
        .nhsCardRowStyle()
}
```

## Accessibility

> The modifier only affects how a view looks. Your content is responsible for its own accessibility, such as labels, traits and tap target sizes.

Reviewer: ```suggestion
The modifier only affects how a view looks. You are responsible for the accessibility of the content used, such as labels, traits and tap target sizes.
```
